rust: kernel: add missing safety comments

DjLogozzo opened the pull request:
Contributes to #351

Added `// SAFETY` docs to `fn proc_handler`
`rust/kernel/sysctl.rs:97:1`
`rust/kernel/sysctl.rs:113:5`
`rust/kernel/sysctl.rs:115:9`
`rust/kernel/sysctl.rs:120:5`
`rust/kernel/sysctl.rs:124:5`
`rust/kernel/sysctl.rs:136:5`
`rust/kernel/sysctl.rs:138:5`

Added `// SAFETY` docs to `fn init_lock`
`rust/kernel/sync/mutex.rs:85:9`

Added `// SAFETY` docs to `fn init_lock`
`rust/kernel/sync/spinlock.rs:144:9`

Added `// SAFETY` docs to `fn init`
`rust/kernel/sync/condvar.rs:135:9`

Added `// SAFETY` docs to `fn read`
`rust/kernel/pages.rs:96:9`

Added `// SAFETY` docs to `fn write`
`rust/kernel/pages.rs:125:9`

Added `// SAFETY` docs to `fn rust_fmt_argument`
`rust/kernel/print.rs:17:1`
`rust/kernel/print.rs:62:5`

Added `// SAFETY` docs to `fn set_param`
`rust/kernel/module_param.rs:63:5`
`rust/kernel/module_param.rs:74:13`
`rust/kernel/module_param.rs:80:17`
`rust/kernel/module_param.rs:83:17`

Added `// SAFETY` docs to `fn get_param`
`rust/kernel/module_param.rs:104:9`
`rust/kernel/module_param.rs:108:9`

Added `// SAFETY` docs to `fn free`
`rust/kernel/module_param.rs:124:9`

Signed-off-by: Dee-Jay Logozzo <git@djl.id.au>

ojeda commented on rust/kernel/module_param.rs:
```diff
@@ -71,11 +71,17 @@ pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
         let arg = if val.is_null() {
             None
         } else {
+            // SAFETY: `val` is a valid CStr by the safety requirements of
+            //         this function
```
Could you please follow a closer style to other files? e.g.

```suggestion
            // SAFETY: `val` is a valid `CStr` by the safety requirements of
            // this function.
```

Also, it is slightly confusing to say `val` is a valid `CStr`, since it is not a `CStr`.

More importantly, we have to show that `val` points to a NUL-terminated string (`val` could have been NULL according to the contract above, but we just checked for it). Also, we should explain why the contents are not mutated for the lifetime (this is called from the C side, so perhaps you need to expand the safety contract to say so).

ojeda commented:
A couple extra notes:
  - I noticed an "atleast" typo in a few places.
  - There is no need to list every single change in the commit message (one can see the contents, after all), but I appreciate you were trying to be thorough :)

DjLogozzo commented:
@ojeda
Thank you for the suggestions, I've implemented the changes you've requested and updated the remaining comments to follow a closer style to the rest of the files.
If there are any more changes you'd like from my PR just let me know

>     * There is no need to list every single change in the commit message (one can see the contents, after all), but I appreciate you were trying to be thorough :)
Yeah, I know it isn't required or even necessary with git, I just wrote them initially to help me with jumping around the files to my changes in my editor when looking at `git log`, however since they were already written in the individual commits on my fork, I though I might as well include them in the squashed PR

Bevisy commented:
Ping @DjLogozzo merge conflicts

ojeda commented:
@Bevisy Given we are now in mainline, if you are still interested in this, I would recommend sending them to the mailing list directly (i.e. those that apply, mainline does not have everything that the `rust` branch had). For that, the changes could be split per-file or similar.
